frontend/src/app/main/data/workspace/tokens/application.cljs

```diff
@@ -31,6 +31,46 @@
 
 (declare token-properties)
 
+;; Match shape by attributes ---------------------------------------------------
+
+(def position-attributes #{:x :y})
+
+(def generic-attributes
+  (set/union ctt/color-keys
+             ctt/stroke-width-keys
+             ctt/rotation-keys
+             ctt/sizing-keys
+             ctt/opacity-keys
+             position-attributes))
+
+(def rect-attributes
+  (set/union generic-attributes
+             ctt/border-radius-keys))
+
+(def frame-attributes
+  (set/union rect-attributes
+             ctt/spacing-keys))
+
+(def text-attributes
+  (set/union generic-attributes
+             ctt/typography-keys
+             ctt/number-keys))
+
+(defn shape-type->attributes
+  [type]
+  (case type
+    :bool    generic-attributes
+    :circle  generic-attributes
+    :rect    rect-attributes
+    :frame   frame-attributes
+    ;; TODO: Groups dont accept tokens yet
+    ;; :group   generic-attributes
```

Comment:
Groups have not any own attributes, and never will (always inherit position and size from content).

So I think that this TODO is not necessary.